TimePicker/tests/TimePicker-specs.js

```diff
@@ -1,195 +1,290 @@
-			base.prop('onChange')({value: 0});
+		const hourPickerUp = screen.getByLabelText('3 hour change a value with up down button').children.item(0);
 
-			const expected = 1;
-			const actual = handleChange.mock.calls.length;
+		userEvent.click(hourPickerUp);
 
-			expect(actual).toBe(expected);
-		}
-	);
+		const expected = 1;
+
+		expect(handleChange).toBeCalledTimes(expected);
+	});
 
 	test('should omit labels when noLabels is true', () => {
-		const subject = mount(
-			<TimePicker hour={1} meridiem={0} meridiems={['am', 'pm']} minute={1} noLabels open order={['h', 'm', 'a']} title="Time" />
+		render(
+			<TimePicker
+				hour={1}
+				meridiem={0}
+				meridiems={['am', 'pm']}
+				minute={1}
+				noLabels
+				open
+				order={['h', 'm', 'a']}
+				title="Time"
+			/>
 		);
 
-		const expected = 3;
-		const actual = subject.find(`.${css.timeComponents}`).children().filterWhere(c => !c.prop('label')).length;
+		const hourLabel = screen.queryByText('hour');
+		const minuteLabel = screen.queryByText('minute');
+		const meridiemLabel = screen.queryByText('AM / PM');
 
-		expect(actual).toBe(expected);
+		expect(hourLabel).toBeNull();
+		expect(minuteLabel).toBeNull();
+		expect(meridiemLabel).toBeNull();
 	});
 
 	test('should create pickers arranged by order', () => {
-		const subject = mount(
-			<TimePicker hour={1} meridiem={0} meridiems={['am', 'pm']} minute={1} open order={['h', 'm', 'a']} title="Time" />
+		render(
+			<TimePicker
+				hour={1}
+				meridiem={0}
+				meridiems={['am', 'pm']}
+				minute={1}
+				open
+				order={['h', 'm', 'a']}
+				title="Time"
+			/>
 		);
-
-		const expected = ['hour', 'minute', 'AM / PM'];
-		const actual = subject.find(`.${css.timeComponents}`).children().map(c => c.prop('label'));
-
-		expect(actual).toEqual(expected);
+		const dateComponent = screen.getByText('minute').parentElement.parentElement;
+
+		const expectedFirst = 'hourComponents';
+		const actualFirst = dateComponent.children.item(0);
+		const expectedSecond = 'minutesComponents';
+		const actualSecond = dateComponent.children.item(1);
+		const expectedThird = 'meridiemComponent';
+		const actualThird = dateComponent.children.item(2);
+
+		expect(actualFirst).toHaveClass(expectedFirst);
+		expect(actualSecond).toHaveClass(expectedSecond);
+		expect(actualThird).toHaveClass(expectedThird);
 	});
 
 	test('should accept a JavaScript Date for its value prop', () => {
-		const subject = mount(
-			<TimePicker open title="Date" value={new Date(2000, 0, 1, 12, 30)} locale="en-US" />
+		render(
+			<TimePicker
+				locale="en-US"
+				open
+				title="Date"
+				value={new Date(2000, 0, 1, 12, 30)}
+			/>
 		);
+		const minutes = screen.getByText('30');
 
-		const minutePicker = subject.find(`.${css.minutesComponents}`).at(0);
+		const expected = 'item';
 
-		const expected = 30;
-		const actual = minutePicker.prop('value');
-
-		expect(actual).toBe(expected);
+		expect(minutes).toBeInTheDocument();
+		expect(minutes).toHaveClass(expected);
```

Comment:
maybe it would be nice to verifiy that it is the minute part of the date. Check that aria-valueText

![image](https://user-images.githubusercontent.com/63335068/127167176-9a84b544-a40d-404c-9ae5-3a1dd11ed0de.png)